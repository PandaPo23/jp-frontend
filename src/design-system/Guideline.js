import React from 'react';

import Absolute from './Absolute';
import ScreenOverlay from './ScreenOverlay';
import Text from './Text';

const Guideline = ({ zIndex = 101, left = '50%', borderStyle = 'dashed' }) => (
  <ScreenOverlay zIndex={zIndex} bg="transparent">
    <Absolute
      width="1px"
      height="100%"
      borderLeft={1}
      borderColor="white"
      left={left}
      borderStyle={borderStyle}
    >
      <Text
        info
        p={2}
        fontSize={4}
        position="absolute"
        top="50%"
        transform="rotate(-90deg)"
      >
        {left}px
      </Text>
    </Absolute>
  </ScreenOverlay>
);

export default Guideline;
