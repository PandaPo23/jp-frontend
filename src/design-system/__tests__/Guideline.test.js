import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Guideline from '../Guideline';

describe('Gridline component', () => {
  it('should render Guideline component', () => {
    const wrapper = mount(<Guideline />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should render guideline with props', () => {
    const localProps = {
      left: '30%',
    };
    const wrapper = mount(<Guideline {...localProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(
      wrapper
        .find(Guideline)
        .first()
        .prop('left')
    ).toBe('30%');
  });
});
