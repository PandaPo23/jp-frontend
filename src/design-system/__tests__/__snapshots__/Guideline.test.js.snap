// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gridline component should render Guideline component 1`] = `
.emotion-2 {
  background-color: transparent;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: 101;
  position: absolute;
}

.emotion-1 {
  width: 1px;
  height: 100%;
  border-color: white;
  border-left: 1px;
  border-style: dashed;
  left: 50%;
  position: absolute;
  position: absolute;
}

.emotion-0 {
  font-size: 24px;
  padding: 8px;
  position: absolute;
  top: 50%;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

<Guideline>
  <ScreenOverlay
    bg="transparent"
    bottom={0}
    left={0}
    position="absolute"
    right={0}
    top={0}
    zIndex={101}
  >
    <div
      className="emotion-2"
    >
      <Absolute
        borderColor="white"
        borderLeft={1}
        borderStyle="dashed"
        height="100%"
        left="50%"
        position="absolute"
        width="1px"
      >
        <div
          className="emotion-1"
        >
          <Text
            fontSize={4}
            info={true}
            p={2}
            position="absolute"
            top="50%"
            transform="rotate(-90deg)"
          >
            <div
              className="emotion-0"
              transform="rotate(-90deg)"
            >
              50%
              px
            </div>
          </Text>
        </div>
      </Absolute>
    </div>
  </ScreenOverlay>
</Guideline>
`;

exports[`Gridline component should render guideline with props 1`] = `
.emotion-2 {
  background-color: transparent;
  bottom: 0;
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: 101;
  position: absolute;
}

.emotion-0 {
  font-size: 24px;
  padding: 8px;
  position: absolute;
  top: 50%;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.emotion-1 {
  width: 1px;
  height: 100%;
  border-color: white;
  border-left: 1px;
  border-style: dashed;
  left: 30%;
  position: absolute;
  position: absolute;
}

<Guideline
  left="30%"
>
  <ScreenOverlay
    bg="transparent"
    bottom={0}
    left={0}
    position="absolute"
    right={0}
    top={0}
    zIndex={101}
  >
    <div
      className="emotion-2"
    >
      <Absolute
        borderColor="white"
        borderLeft={1}
        borderStyle="dashed"
        height="100%"
        left="30%"
        position="absolute"
        width="1px"
      >
        <div
          className="emotion-1"
        >
          <Text
            fontSize={4}
            info={true}
            p={2}
            position="absolute"
            top="50%"
            transform="rotate(-90deg)"
          >
            <div
              className="emotion-0"
              transform="rotate(-90deg)"
            >
              30%
              px
            </div>
          </Text>
        </div>
      </Absolute>
    </div>
  </ScreenOverlay>
</Guideline>
`;
